src/Directory.Build.props

```diff
@@ -41,17 +42,7 @@
 
     <UNO_UWP_BUILD>true</UNO_UWP_BUILD>
     <DefineConstants Condition="'$(UNO_UWP_BUILD)'!='true'">$(DefineConstants);HAS_UNO_WINUI</DefineConstants>
-    <DefineConstants Condition="
-		$(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('monoandroid'))
-		OR $(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('net6.0-android'))
-		OR $(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('net7.0-android'))
-		">$(DefineConstants);TARGET_FRAMEWORK_OVERRIDE_ANDROID</DefineConstants>
-
-    <DefineConstants Condition="
-		$(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('xamarinios'))
-		OR $(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('net6.0-ios'))
-		OR $(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('net7.0-ios'))
-		">$(DefineConstants);TARGET_FRAMEWORK_OVERRIDE_IOS</DefineConstants>
```

Comment:
`TARGET_FRAMEWORK_OVERRIDE_ANDROID` and `TARGET_FRAMEWORK_OVERRIDE_IOS` are only used in https://github.com/unoplatform/uno/blob/55c0447b1c5367032fa81695f8c2a2684292e6cc/src/SamplesApp/SamplesApp.UITests/Constants.cs#L17-L25

So moved this to the test project, and replaced the two `StartsWith` with a single `EndsWith('-android')`